libtransmission/inout.cc

```diff
@@ -110,16 +112,11 @@ void readOrWriteBytes(
 {
     TR_ASSERT(file_index < tor->file_count());
 
-    bool const do_write = io_mode == IoMode::Write;
     auto const file_size = tor->file_size(file_index);
+    bool const do_write = io_mode == IoMode::Write || file_size == 0U;
```

Comment:
If ```io_mode == IoMode::Read``` I'd expect ```readOrWriteBytes()``` will only read. But if ```file_size == 0``` ```do_write``` becomes ```true```, which looks for me unexpected.

Reply:
Resolving since this is no longer relevant now that this change was reverted.